sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_discovery.py

```diff
@@ -0,0 +1,97 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+import time
+from urllib.parse import urlparse
+import dns.resolver
+from dns.resolver import NXDOMAIN, YXDOMAIN, LifetimeTimeout, NoNameservers  # cspell:disable-line
+from dns.rdatatype import SRV  # cspell:disable-line
+from ._constants import HTTPS_PREFIX
+
+request_retry_period = 5  # seconds
+
+
+class SRVRecord:
+    priority: int
+    weight: int
+    port: int
+    target: str
+
+    def __init__(self, answer):
+        self.priority = answer.priority
+        self.weight = answer.weight
+        self.port = answer.port
+        self.target = str(answer.target).rstrip(".")
+
+
+def find_auto_failover_endpoints(endpoint: str, replica_discovery_enabled: bool):
+    if replica_discovery_enabled:
```

Comment:
Did you mean to check the opposite? If replica discovery disabled, return empty list